### DISCUSSION TITLE
Use of this library with IsaacGym

### DISCUSSION THREAD

**Discussion by ismarou** [NVlabs/curobo#8] [Software Q&A] [answered]
Hello, 

This library is awesome!
How can I wrap these IsaacSim functionalities in order to use them in an IsaacGym environment? Do I need anything special?

Thank you in advance,

Best,
Isidoros

**Comment by balakumar-s** [comment]
Thanks! The best way to integrate into Isaac Gym would be to use the python examples as a starting point: https://curobo.org/source/getting_started/2a_python_examples.html

Since Isaac Gym does not use a usd stage to represent the world, you would have to read the pose of all objects in your gym environment and add them to CuRobo's worldconfiguration. 

Is there a specific functionality you are looking at using?

**Reply by pushkalkatara** [reply]
Hi, Thanks for the amazing library on motion planning!

I am trying to use CuRobot with other simulators, specifically IssacGym.

I have a scene URDF and want to load this scene into CuRobot WorldConfig. What steps should I follow for this conversion?

For IssacSim, UsdHelper is provided, is there a specific tool for IssacGym or URDF/XML scenes?

https://github.com/NVlabs/curobo/discussions/195#discussion-6406506

**Comment by LyuJZ** [comment]
Hi, great work!

Isaac Gym is a lightweight yet powerful tool for reinforcement learning research. However, the classic functions for robot control, like advanced IK solvers, motion-planning algorithms, are really missing. 

It would be really helpful for the community to have some simple examples or tutorials about using curobo with Isaac Gym.

**Reply by balakumar-s** [reply, marked as answer]
We don't have the resources to build examples for Isaac Gym. The hope with the provided examples in python and isaac sim is to show how anyone can integrate curobo into their own pipeline/simulators.

**Comment by frederiknolte** [comment]
May I ask if you have been able to pair curobo with IsaacGym? Is it possible to do motion planning for multiple environments in parallel?

**Reply by scharalambous-sony** [reply]
There is an example called batch_motion_gen_reacher.py
I believe it is relevant to what you are looking for
